src/sprintf.js

```diff
@@ -41,7 +41,7 @@
                 if (ph.keys) { // keyword argument
                     arg = argv[cursor]
                     for (k = 0; k < ph.keys.length; k++) {
-                        if (!arg.hasOwnProperty(ph.keys[k])) {
+                        if (arg[ph.keys[k]] === undefined) {
```

Comment:
I'm not sure, but to preserve backwards compatibility, `!(ph.keys[k] in arg)` might be a better approach